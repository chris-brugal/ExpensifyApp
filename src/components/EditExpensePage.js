import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startEditExpenses, startRemoveExpenses} from '../actions/expenses';


export class EditExpensePage extends React.Component {
  onSubmit = (expense)=> {
    //dispatch the action to edit the expense
    //redirect to the dashboard
    this.props.startEditExpenses(this.props.expense.id, expense);
    this.props.history.push('/');
    //console.log('updated', expense);
  }

  onClick = ()=>{
    this.props.startRemoveExpenses({id: this.props.expense.id});
    this.props.history.push("/");
  }

  render(){
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
              <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm 
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button-remove" onClick={this.onClick}>Remove Expense</button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
   return{
     expense: state.expenses.find((expense) => expense.id === props.match.params.id)
   };
}

const mapDispatchToProps = (dispatch, props) => {
  return{
    startEditExpenses: (id, expense) => dispatch(startEditExpenses(id, expense)),
    startRemoveExpenses: (data) => dispatch(startRemoveExpenses(data))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
