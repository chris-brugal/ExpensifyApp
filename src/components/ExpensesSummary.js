import React from 'react';
import {connect} from 'react-redux';
import selectedExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral'

const ExpenseSummary = (props) =>{
    console.log(props);
    return(
        <div> 
            {
                props.expenses.length >= 1? 
                    (props.expenses.length === 1? 
                        <h1>Viewing 1 expense totalling {numeral(expensesTotal(props.expenses)/100).format('$0,0.00')}</h1> 
                        :
                        <h1>Viewing {props.expenses.length} expenses totalling {numeral(expensesTotal(props.expenses)/100).format('$0,0.00')}</h1>)
                : 
                (<h1>Viewing 0 expenses totalling $0.00</h1>)
            }
        </div>
    )
};

const mapStateToProps = (state) =>{
    //the filtered expenses are being shown now with the call to the selected expenses func
    return{
        expenses: selectedExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseSummary);