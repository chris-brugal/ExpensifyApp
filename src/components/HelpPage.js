import React from 'react';

const HelpPage = () =>{
    return(
        <div> this is my help page</div>
    );
}
export default HelpPage;