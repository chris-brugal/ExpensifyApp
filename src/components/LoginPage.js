import React from 'react';
import {connect} from 'react-redux';
import {startLogin, startFacebookLogin} from '../actions/auth';

export const LoginPage = ({startLogin, startFacebookLogin}) => {
    return(
        <div className="box-layout"> 
            <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify</h1>
                <p>It's time to get your expenses mobile.</p>
                <button onClick={startLogin} className="button">Login with Google</button>
                <button onClick={startFacebookLogin} className="button--bottom">Login with Facebook</button>
            </div>
        </div>
    );
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);