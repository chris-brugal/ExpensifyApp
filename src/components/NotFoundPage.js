import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () =>{
    //link is used internally in order to take adv of client side routing. its fast. not full reload
    return(
        <div>
         404: <Link to="/">Go home</Link>
        </div>
    );
}
export default NotFoundPage;