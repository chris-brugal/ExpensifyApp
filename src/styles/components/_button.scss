.button{
    color: white;
    background: $blue;
    border: none;
    display: inline-block;
    font-weight: lighter;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link{
    background: none;
}

.button-remove{
    @extend .button;
    background: #888;
}

.button--bottom{
    @extend .button;
   margin-top: 10px;
}