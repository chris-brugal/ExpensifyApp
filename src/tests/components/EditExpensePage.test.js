import React from 'react';
import expenses from '../fixtures/expenses';
import {shallow} from 'enzyme'; 
import {EditExpensePage} from '../../components/EditExpensePage';

//refactor edit expense to be a class based component
//setup mapDispatchToProps editExp and remExp
let editExpense, history, wrapper, removeExpense ;
beforeEach(() => {
    removeExpense = jest.fn();
    editExpense = jest.fn();
    history = {push: jest.fn()};
    wrapper = shallow(<EditExpensePage history={history} editExpense={editExpense} removeExpense={removeExpense} expense={expenses[2]}/>);
});

//should render Edit (snapshot)
test('Should render edit', () => {
    expect(wrapper).toMatchSnapshot();    
});

//should handle edit exp (spies)
test('should handle edit exp', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses[2]);

});

//should handle remove exp (spies)
test('should handle remove exp', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({id: expenses[2].id});
});
