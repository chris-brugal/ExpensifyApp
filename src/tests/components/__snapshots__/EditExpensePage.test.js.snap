// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render edit 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 100000,
        "createdAt": 345600000,
        "description": "Card Payment",
        "id": 3,
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
